### PR TITLE
PAPP-28878 Ighelani azuredevops update

### PR DESCRIPTION
**Pull request by ighelani-crest**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [ ] Testing of all the changes has been performed (for bug fixes / features)
- [ ] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [ ] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [ ] Verify all checks are passing.
- [ ] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [ ] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Security Considerations (REQUIRED)
- [ ] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers), 
  please document them in the `readme.html`.
- [ ] If this is a new connector or you are adding new actions
    - [ ] Please document in the `readme.html` all methods (eg, OAuth) used to authenticate 
      with the service that the connector is integrating with.
    - [ ] If any actions are unable to run on SOAR Cloud, please document this in the `readme.html`.
- [ ] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [ ] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through 
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [ ] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

  

## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,32 +1,185 @@
...
+        > &state={state}  
+        > &scope={scope}  
+        > &redirect_uri={callback URL}
+
+    | Parameter     | Type   | Notes                                                                                                                        |
+    |---------------|--------|------------------------------------------------------------------------------------------------------------------------------|
+    | client_id     | GUID   | he ID assigned to your app when it was registered.                                                                           |
+    | response_type | string | Assertion                                                                                                                    |
+    | state         | string | Can be any value. Typically a generated string value that correlates the callback with its associated authorization request. |
+    | scope         | string | Scopes registered with the app. Space separated.                                                                             |
```
[the remark this answers is not in the document]
Comment (ighelani-crest) [reply README.md:77]: changed the doc

File: README.md [hunk 2]
```diff
@@ -1,32 +1,185 @@
...
+        > https://app.vssps.visualstudio.com/oauth2/authorize  
+        > ?client_id={app ID}  
+        > &response_type={Assertion}  
+        > &state={state}  
+        > &scope={scope}  
+        > &redirect_uri={callback URL}
+
+    | Parameter     | Type   | Notes                                                                                                                        |
+    |---------------|--------|------------------------------------------------------------------------------------------------------------------------------|
+    | client_id     | GUID   | he ID assigned to your app when it was registered.                                                                           |
```
[the remark this answers is not in the document]
Comment (ighelani-crest) [reply README.md:74]: changed the doc

File: README.md [hunk 3]
```diff
@@ -1,32 +1,185 @@
...
+    URL, including a short-lived authorization code and the state value provided in the
+    authorization URL:
+
+    > https://fabrikam.azurewebsites.net/myapp/oauth-callback  
+    > ?code={authorization code}  
+    > &state=User1
+
+    ## Configure the Azure Devops Splunk SOAR app Asset
+
+    When creating an asset for the **Azure Devops** app,go to asset settings and place the
```
[the remark this answers is not in the document]
Comment (ighelani-crest) [reply README.md:90]: changed the doc

File: README.md [hunk 4]
```diff
@@ -1,32 +1,185 @@
...
+    **Application ID** of the app created during the previous step in the **Client ID** field and
+    place the Client Secret generated during the previous step in the **Client Secret** field. Then,
+    after filling out the **The name of the Azure DevOps organization** and **Project name** fields
+    , click **SAVE** .  
+      
+    After saving, a new field will appear in the **Asset Settings** tab. Take the URL found in the
+    **POST incoming for Azure DevOps to this location** field and place it in the **Application
+    website** field mentioned in a previous step.Add the same url to the **Authorization callback
```
[the remark this answers is not in the document]
Comment (ighelani-crest) [reply README.md:98]: Changed the doc

File: README.md [hunk 5]
```diff
@@ -1,32 +1,185 @@
...
+    Azure Devops environment. After logging in, review the requested permissions listed, then click
+    **Accept** . Finally, close that tab. The test connectivity window should show a success.  
+      
+    The app should now be ready to use.  
+      
+    If username and password is entered than priority will be given to the basic auth then
+    Interactive Auth.  
+      
+    We have tested all the action for the api version 7.0 but it might be supported in another
+    versions as well.
```
[the remark this answers is not in the document]
Comment (ighelani-crest) [reply README.md:121]: changed the doc

File: azuredevops.json [hunk 6]
```diff
@@ -103,331 +131,2267 @@
...
+                "type": "table"
+            },
+            "versions": "EQ(*)"
+        },
+        {
+            "action": "delete user",
+            "identifier": "delete_user",
+            "description": "Delete a user",
+            "type": "generic",
+            "read_only": true,
```
Comment (alexa-phantom) [inline azuredevops.json:1920]:
```suggestion
            "read_only": false,
```

**Review by alexa-phantom — CHANGES_REQUESTED** [review]

**Review by alexa-phantom — APPROVED** [review]

**Review by ricardo-phantom — APPROVED** [review]

**Review by ishans-crest — APPROVED** [review]

**Pull request merged**